eth/consensus/clique/_utils.py

```diff
@@ -0,0 +1,154 @@
+import time
+from typing import (
+    Iterable,
+)
+
+from eth.abc import (
+    BlockHeaderAPI,
+)
+from eth.constants import (
+    EMPTY_UNCLE_HASH,
+    ZERO_ADDRESS,
+    ZERO_HASH32,
+)
+from eth_keys import keys
+from eth_keys.datatypes import PrivateKey
+from eth_typing import (
+    Address,
+    Hash32,
+)
+from eth_utils import (
+    encode_hex,
+    to_list,
+    ValidationError,
+)
+
+from .constants import (
+    COMMON_ADDRESS_LENGTH,
+    NONCE_AUTH,
+    NONCE_DROP,
+    SIGNATURE_LENGTH,
+    VANITY_LENGTH,
+)
+from .datatypes import (
+    Snapshot,
+)
+
+
+@to_list
```

Comment:
Is this intentionally using a mutable data structure?

Reply:
Nope, just a left over from the make-it-work stages. I'll change that to `to_tuple`